src/authorization/AuxiliaryAuthorizer.ts

```diff
@@ -0,0 +1,55 @@
+import type { AuxiliaryManager } from '../ldp/auxiliary/AuxiliaryManager';
+import { getLoggerFor } from '../logging/LogUtil';
+import { NotImplementedHttpError } from '../util/errors/NotImplementedHttpError';
+import type { AuthorizerArgs } from './Authorizer';
+import { Authorizer } from './Authorizer';
+
+/**
+ * An authorizer for auxiliary resources.
+ * Calls the source authorizer with potentially updated permissions
+ * and the associated resource identifier.
+ */
+export class AuxiliaryAuthorizer extends Authorizer {
+  protected readonly logger = getLoggerFor(this);
+
+  private readonly authorizer: Authorizer;
+  private readonly manager: AuxiliaryManager;
+
+  public constructor(authorizer: Authorizer, manager: AuxiliaryManager) {
+    super();
+    this.authorizer = authorizer;
+    this.manager = manager;
+  }
+
+  public async canHandle(input: AuthorizerArgs): Promise<void> {
+    const newArgs = this.getSafeAssociatedArgs(input);
+    return this.authorizer.canHandle(newArgs);
+  }
+
+  public async handle(input: AuthorizerArgs): Promise<void> {
+    const newArgs = this.getAssociatedArgs(input);
+    this.logger.debug(`Redirecting auth request for ${input.identifier.path} to ${newArgs.identifier.path}`);
```

Comment:
To avoid HTTP redirection confusion:
```suggestion
    this.logger.debug(`Checking authorization for ${input.identifier.path} on ${newArgs.identifier.path}`);
```